types/ember-data/index.d.ts

```diff
@@ -742,7 +769,7 @@ export namespace DS {
      * addon author to perform meta-operations on a belongs-to
      * relationship.
      */
-    class BelongsToReference {
+    class BelongsToReference<T extends Model> {
```

Comment:
To avoid making this a breaking change for users who already consume the `BelongsToReference` type, I think we need a default type param `= Model`